### PR TITLE
Anomali ThreatStream v2 get-indicators limitation bug

### PR DESCRIPTION
**Pull request by ChanochShayner**
… return more than 1000 indicators.

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/29858

## Description
Fixed an issue where **threatstream-get-indicators** command does not return more than 1000 indicators.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py [hunk 1]
```diff
@@ -891,13 +891,25 @@ def get_indicators(**kwargs):
...
-        params = build_params(q=kwargs['query'], limit=kwargs.get('limit', 20))
+        params = build_params(q=kwargs['query'], limit=limit)
     else:
         params = build_params(**kwargs)
 
     iocs_list = http_request("GET", "v2/intelligence/", params=params).get('objects', None)
 
+    if limit > 1000:
+        limit -= 1000
+        offset = 1000
```
Comment (ronykoz) [inline Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:904]: you should handle numbers like 20000 as well

Comment (ChanochShayner) [reply Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py]: done

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py [hunk 2]
```diff
@@ -892,18 +892,29 @@ def get_indicators(**kwargs):
         Returns filtered indicators by parameters from ThreatStream.
         By default the limit of indicators result is set to 20.
     """
+    limit = kwargs['limit'] = int(kwargs.get('limit', 20))
+    offset = kwargs['offset'] = 0
```
Comment (ronykoz) [inline Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:896]: is offset always 0?

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py [hunk 3]
```diff
@@ -892,18 +892,29 @@ def get_indicators(**kwargs):
...
+        kwargs.pop('query', None)
+    params = build_params(**kwargs)
     iocs_list = http_request("GET", "v2/intelligence/", params=params).get('objects', None)
-
     if not iocs_list:
         demisto.results('No indicators found from ThreatStream')
         sys.exit()
-
     iocs_context = parse_indicators_list(iocs_list)
+    if limit > 1000:
```
Comment (ronykoz) [inline Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:906]: not sure how are you dealing with the first request here, it should be up to 1000 indicators, no?

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py (hunk 2 above)
Comment (ChanochShayner) [reply Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:896]: Yes, there is no such argument for the user in this command, and it comes in handy only when requesting more than 1000 indicators.

File: Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py (hunk 3 above)
Comment (ChanochShayner) [reply Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:906]: Between lines 901 and 905 we handle a standard user request (which is usually less than 1000)
If the request is for more than 1000 then we execute this request again and again each time with an offset of the indicators we have already received until reaching the number the user wanted.

Comment (ronykoz) [reply Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py:906]: shouldn't we update the limit to 1000 in that case?

**Review by ronykoz — APPROVED** [review]
Add a comment regarding the 1000 as we talked about over slack

**Pull request merged**